db/users.go

```diff
@@ -145,6 +140,31 @@ func (dbc *DatabaseContext) UpdatePrincipal(ctx context.Context, updates *auth.P
 			if updates.ExplicitRoleNames != nil && !updatedExplicitRoles.Equals(updates.ExplicitRoleNames) {
 				changed = true
 			}
+
+			if updates.OIDCIssuer != nil && *updates.OIDCIssuer != user.OIDCIssuer() {
```

Comment:
If the issuer changes, but updates.OIDCRoles (or updates.OIDCChannels) is nil, should we be setting those values to nil?

Reply:
Not sure I follow - `updates.OIDCRoles` being nil means "don't change this", am I missing something?

Reply:
As discussed, we just want a check that the OIDCIssuer/OIDCRoles/OIDCChannels are all or none on the incoming updates.